scripts/blt/composer-version-convert.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Usage:
+ * php -f convert-to-drupal-org-packagist.php $(pwd)/composer.json
+ */
+
+$composer_json_filepath = $argv[1];
+$composer_json = json_decode(file_get_contents($composer_json_filepath), TRUE);
+
+// Update version constraints in require.
+foreach ($composer_json['require'] as $package_name => $version_constraint) {
+  if (strstr($package_name, 'drupal/')) {
+    $composer_json['require'][$package_name] = convert_version_constraint($version_constraint);
+  }
+}
+
+// Change drupal/lightning to acquia/lightning.
+$composer_json['require']['acquia/lightning'] = '^2';
+unset($composer_json['require']['drupal/lightning']);
+
+// Update version constraints in require-dev.
+foreach ($composer_json['require-dev'] as $package_name => $version_constraint) {
+  if (strstr($package_name, 'drupal/')) {
+    $composer_json['require-dev'][$package_name] = convert_version_constraint($version_constraint);
+  }
+}
+
+// Remove https://packagist.drupal-composer.org.
+foreach ($composer_json['repositories'] as $key => $repository) {
+  if ($repository['url'] == 'https://packagist.drupal-composer.org') {
+    unset($composer_json['repositories'][$key]);
+  }
+}
+
+// Add https://packages.drupal.org/8.
+$composer_json['repositories']['drupal'] = [
+  'type' => 'composer',
+  'url' => 'https://packages.drupal.org/8',
+];
+
+$output = json_encode($composer_json, JSON_PRETTY_PRINT|JSON_UNESCAPED_SLASHES);
+print $output;
+
+function convert_version_constraint($version_constraint) {
+
+  // 8.* => *
+  if (preg_match('/8\.\*$/', $version_constraint)) {
```

Comment:
There's no anchor at the beginning, so won't this also match something like 1.8.*?